api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -405,6 +365,43 @@ private DataSetRequest generateDataSetRequestFromPreviewRequest(
     return ResponseEntity.ok(previewQueryResponse);
   }
 
+  private void addFieldValuesFromBigQueryToPreviewList(List<DataSetPreviewValueList> valuePreviewList, FieldValueList fieldValueList) {
+    IntStream.range(0, fieldValueList.size())
+        .forEach(
+            columnNumber -> {
+              valuePreviewList
+                  .get(columnNumber)
+                  .addQueryValueItem(
+                      Optional.ofNullable(fieldValueList.get(columnNumber).getValue())
+                          .orElse("")
+                          .toString());
+            });
+  }
+
+  private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewList, Field fields) {
```

Comment:
nit: A short comment describing what this function does would help here. It could also be unit tested if there's time.

Also, why is `fields` plural here?